src/librustc/middle/resolve_lifetime.rs

```diff
@@ -89,9 +89,9 @@ struct LifetimeContext<'a, 'tcx: 'a> {
 
 #[derive(PartialEq, Debug)]
 enum ScopeChain<'a> {
-    /// EarlyScope(['a, 'b, ...], s) extends s with early-bound
-    /// lifetimes.
-    EarlyScope(&'a [hir::LifetimeDef], Scope<'a>),
+    /// EarlyScope(['a, 'b, ...], start, s) extends s with early-bound
+    /// lifetimes, with consecutive indexes from `start`.
```

Comment:
Nit: This phrasing isn't very clear to me. i think it means "there are `start` types in front"?